go/libraries/doltcore/sqle/dprocedures/dolt_branch.go

```diff
@@ -53,7 +53,9 @@ func doDoltBranch(ctx *sql.Context, args []string) (int, error) {
 		return 1, fmt.Errorf("Empty database name.")
 	}
 
-	apr, err := cli.CreateBranchArgParser().Parse(args)
+	ap := cli.CreateBranchArgParser()
+	ap.SupportsFlag(cli.RemoteParam, "r", "Delete a remote tracking branch.")
```

Comment:
Looks like this is using a different ArgParser than the one in branch.go. Is there a reason for that? Should we consolidate? At the very lest, add the -r flag to the original definition. Its weird to add it here.

Reply:
The reason is that `dolt_branch.go` is only for actions that modify branches, not ones that only list branches.

The common flags are defined in CreateBranchArgParser.

Both versions have a `-r` flag, but their descriptions are different.

I added comments to explain.